sycl/include/CL/sycl/types.hpp

```diff
@@ -317,13 +317,32 @@ template <typename T, int N> struct VectorAlignment {
 
 } // namespace detail
 
+#if defined( __GNUC__ )
+#define ALIGNAS(N) alignas(N)
+#elif defined( _WIN32) && (_MSC_VER)
+// MSVC Compiler doesn't allow using of function arguments with alignment requirements.
+// MSVC Compiler Error C2719: 'parameter': formal parameter with 
+// __declspec(align('#')) won't be aligned. The align __declspec modifier
+// is not permitted on function parameters. Function parameter alignment
+// is controlled by the calling convention used.
+// For more information, see Calling Conventions
+// (https://docs.microsoft.com/en-us/cpp/cpp/calling-conventions).
+// For information on calling conventions for x64 processors, see 
+// Calling Convention
+// (https://docs.microsoft.com/en-us/cpp/build/x64-calling-convention).
+#define ALIGNAS(x)
+#else
+#warning  Need to implement some method to align data here
```

Comment:
"some method"?
Could you specify what is needed?

Reply:
Hm, a check that new compiler can apply 128 alignment for a type is needed, if yes then allow use alignas, otherwise, rely on the compiler rules

Reply:
This data must be in the message, current warning is quite useless for developers who will hit this case.

Reply:
Updated, please take a look.